src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,573 @@
+            filter_bnds (numpy array of shape (num_filters) x 2): limits of each filter, of the form [[f_1^min, f_1^max], ..., [f_n^min, f_n^max]]. If one of the values is numpy.nan, it can be computed from the dataset with the fit() method.
+            resolutions (numpy array of shape num_filters containing integers): resolution of each filter function, ie number of intervals required to cover each filter image. If None, it is estimated from data.
+            gains (numpy array of shape num_filters containing doubles in [0,1]): gain of each filter function, ie overlap percentage of the intervals covering each filter image.
+            clustering (class): clustering class (default sklearn.cluster.DBSCAN()). Common clustering classes can be found in the scikit-learn library (such as AgglomerativeClustering for instance). If None, it is set to hierarchical clustering, with scale estimated from data.
+            N (int): subsampling iterations (default 100) for estimating scale and resolutions. Used only if clustering or resolutions = None. See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            beta (double): exponent parameter (default 0.) for estimating scale and resolutions. Used only if clustering or resolutions = None. See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            C (double): constant parameter (default 10.) for estimating scale and resolutions. Used only if clustering or resolutions = None. See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            input_name (string): name of dataset. Used when generating plots.
+            cover_name (string): name of cover. Used when generating plots.
+            color_name (string): name of color function. Used when generating plots.
+            verbose (bool): whether to display info while computing.
+        """
+        self.filters, self.filter_bnds, self.resolutions, self.gains, self.colors, self.clustering = filters, filter_bnds, resolutions, gains, colors, clustering
+        self.input_type, self.mask, self.N, self.beta, self.C = input_type, mask, N, beta, C
+        CoverComplexPy.__init__(self, input_name, cover_name, color_name, verbose)
+
+    def estimate_scale(self, X, N=100, inp="point cloud", beta=0., C=10.):
+        """
+        Compute estimated scale of a point cloud or a distance matrix.
+
+        Parameters:
+            X (numpy array of shape (num_points) x (num_coordinates) if point cloud and (num_points) x (num_points) if distance matrix): input point cloud or distance matrix.
+            N (int): subsampling iterations (default 100). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            inp (string): either "point cloud" or "distance matrix". Type of input data (default "point cloud").
```

Comment:
Isn't that redundant with `self.input_type`?

Reply:
Indeed. Fixed.